main.rb

```diff
@@ -4,25 +4,43 @@
 require 'sinatra/reloader'
 require 'json'
 require 'securerandom'
+require 'pg'
 
 # 共通処理
 helpers do
-  # jsonファイルを作成日順に並び替えて、ファイルの中身をhashに変換する処理
-  def fetch_memos_from_json_file
-    json_files = Dir.glob('json/*').sort_by { |file| File.birthtime(file) }
-    json_files.map { |file| JSON.parse(File.read(file)) }
-  end
-
   # XSS対策
   def escape(text)
     Rack::Utils.escape_html(text)
   end
 
-  # hashからjson形式に変換する処理
-  def write_to_json_file(hash)
-    File.open("json/#{hash['id']}.json", 'w') do |file|
-      JSON.dump(hash, file)
-    end
+  # DB接続
+  def connection
+    PG.connect(dbname: 'sinatra_memo_db')
```

Comment:
いいですね！
さらに以下のようにしておくと、connection を使いまわせるようになります。
DB 接続にかかわらず使える技なので覚えておいていただけたら嬉しいです〜 ☺️

```suggestion
    @connection ||= PG.connect(dbname: 'sinatra_memo_db')
```

Reply:
この演算子すごい便利ですねー！！😆
左辺の`@connection`が偽（nilやfalse）であるからできる処理だと理解しました！

教えて頂きありがとうございます🙇‍♂️